src/components/CouponDetails/index.jsx

```diff
@@ -221,7 +228,8 @@ class CouponDetails extends React.Component {
       remainingUses -= redemptions.num_assignments;
     }
 
-    return (
+    // Don't show `Assign` button for an unavailable coupon
+    return couponAvailable ? (
```

Comment:
The conditional check on `couponAvailable` can probably occur higher up in this function, e.g.:

```
getActionButton(code) {
  const {
    couponData: {
      id,
      errors,
      available: couponAvailable,
      title: couponTitle,
    },
  } = this.props;

  // ...

  let remainingUses = redemptions.total - redemptions.used;

  if (!couponAvailable) {
    return null;
  }

  // ...

  return (
    <Button /> // assign button
  );
}
```

Reply:
@adamstankiewicz As per ticket we only want to disable/hide the `Assign` button for coupons which are not available.

**Available Coupon:** `coupon start datetime < current datetime < coupon end datetime`